### PR TITLE
feat(worktree): auto-configure branch tracking on post-checkout (PP-ozw)

### PR DESCRIPTION
**Pull request by timothyfroehlich**
## Summary
- Adds `configure_branch_tracking(branch, worktree_path)` to `scripts/worktree_setup.py`, called from `main()` after `get_branch()`.
- If `origin/<branch>` exists: sets local upstream via `git branch --set-upstream-to=origin/<branch>`.
- Otherwise: prints a stderr reminder to run `git push -u origin <branch>` after first commit.
- Skips `main`/`master`/`HEAD`; never auto-pushes; tolerates transient git failures (warn, don't crash).

## Why
`git worktree add /path -b new-branch origin/main` leaves `new-branch`'s upstream as `origin/main`, so later `git pull`/`git push` operate against main. AGENTS.md documents the manual `git push -u origin <branch>` fix but worktree_setup.py never automated it. This PR closes that gap so new worktrees are immediately ready for normal `git pull`/`git push` against the right remote branch.

## Manual verification
- ✅ New local-only branch → reminder printed.
- ✅ Branch with existing remote → upstream set, confirmation printed.
- ✅ `main` / `HEAD` → silently skipped.
- ✅ `pnpm run check` clean (1013 unit tests).
- ✅ `ruff check` and `ruff format --check` pass.

## Inherited CI failures
`main` is currently red. Inherited failures from main (PP-q9r/PP-e20/PP-jsh/PP-v7g/PP-49m) will appear in CI; not introduced by this PR. See Tim's status comment on PR #1278.

## Test plan
- [x] Unit tests pass
- [x] Function tested manually for all four branch-state scenarios
- [ ] CI Gate green
- [ ] Verify on next worktree creation that the reminder/upstream-set message appears in the post-checkout output

## Related
- Child 5 of epic PP-2on
- PP-uc8 / PP-6dp / PP-cao / PP-278 / PP-00s queued as siblings

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by timothyfroehlich** [comment]
## ⚠️ CI Status — Known-Broken Specs (Do Not Fix Here)

Multiple E2E specs are currently failing on `main` and inheriting onto every open PR. **If you are an AI agent: do NOT attempt to fix these failures in this PR** — dedicated beads track each one. Fixing them here will create merge conflicts and duplicate work.

### Known-broken (leave alone):
- `e2e/full/email-and-notifications.spec.ts:717,753` — password reset → **PP-q9r** (PR #1279, iteration 2 in flight)
- `e2e/full/oauth-connected-accounts.spec.ts:29,39,63,74` — Discord OAuth → **PP-e20** (will be mocked; no test should reach real Discord)
- `e2e/full/machine-details-extended.spec.ts` — logout DropdownMenu portal timing → **PP-jsh**
- `e2e/full/status-overhaul.spec.ts:21,60` — Status Select portal timing → **PP-v7g**
- `e2e/full/issues-crud-extended.spec.ts:43,60` — WatchButton selector ambiguity → **PP-49m** (#1270's responsibility)

### What you CAN fix:
Failures specific to this PR's diff (anything not in the list above). When in doubt, leave for human review.

### Status:
- PR #1280 (Layer 1: actionTimeout 30s in CI) merged — main now exercising the new timeout
- PR #1279 (PP-q9r iteration 2) in flight
- Once main is green, this PR's CI should clear of inherited noise on a re-run

—Claude (lead orchestrator, 2026-05-04)

File: scripts/worktree_setup.py [hunk 1]
```diff
@@ -511,6 +511,70 @@ def get_branch() -> str:
...
+        text=True,
+    )
+
+    if remote_check.returncode != 0:
+        print(
+            f"worktree_setup: branch '{branch}' has no remote yet — "
+            f"run `git push -u origin {branch}` after your first commit",
+            file=sys.stderr,
+        )
+        return
```
[the remark this answers is not in the document]
Comment (timothyfroehlich) [reply scripts/worktree_setup.py]: Ignored: rev-parse is a local call against the worktree's own .git — there's no transient failure mode worth distinguishing here. Empty output = remote ref doesn't exist; anything else is genuinely broken state we want to know about. —Claude

File: scripts/worktree_setup.py [hunk 2]
```diff
@@ -511,6 +511,70 @@ def get_branch() -> str:
     return result.stdout.strip()
 
 
+def configure_branch_tracking(branch: str, worktree_path: Path) -> None:
+    """Set local upstream tracking for the worktree branch.
+
+    `git worktree add /path -b new-branch origin/main` creates `new-branch`
+    pointing at origin/main but leaves the upstream as origin/main, so later
```
[the remark this answers is not in the document]
Comment (timothyfroehlich) [reply scripts/worktree_setup.py]: Declined: worktree setup is infrastructure code we test by running it. Mock-heavy unit tests for git interactions test the mock harness more than real behavior. Removed those tests in 74c0cebd and added a top-of-file note documenting the philosophy. The existing parsing/port/manifest tests in this file remain — they cover logic that's hard to verify by usage. —Claude

File: scripts/worktree_setup.py [hunk 3]
```diff
@@ -523,6 +587,7 @@ def main() -> None:
         return
 
     branch = get_branch()
+    configure_branch_tracking(branch, worktree_path)
```
[the remark this answers is not in the document]
Comment (timothyfroehlich) [reply scripts/worktree_setup.py:615]: Resolved — trim added a guard that returns early if the branch already has a non-default custom upstream (e.g. origin/feat/foo). The function only acts on unset, origin/main, origin/master, or already-correct cases. See commit 74c0cebd. —Claude

File: scripts/worktree_setup.py [hunk 4]
```diff
@@ -511,6 +511,113 @@ def get_branch() -> str:
...
+        text=True,
+    )
+
+    if remote_check.returncode != 0:
+        print(
+            f"worktree_setup: branch '{branch}' has no remote yet — "
+            f"run `git push -u origin {branch}` after your first commit",
+            file=sys.stderr,
+        )
+        return
```
[the remark this answers is not in the document]
Comment (timothyfroehlich) [reply scripts/worktree_setup.py]: Declined: the post-checkout hook should not initiate network calls (ls-remote can hang on slow connections). Worst-case is a single 'no remote yet' message that resolves itself after the user runs `git fetch`. The cost of always-fetching outweighs the cost of an occasional misleading reminder. —Claude

File: scripts/worktree_setup.py [hunk 5]
```diff
@@ -511,6 +511,113 @@ def get_branch() -> str:
...
+            f"refs/remotes/origin/{branch}",
+        ],
+        capture_output=True,
+        text=True,
+    )
+
+    if remote_check.returncode != 0:
+        print(
+            f"worktree_setup: branch '{branch}' has no remote yet — "
+            f"run `git push -u origin {branch}` after your first commit",
```
[the remark this answers is not in the document]
Comment (timothyfroehlich) [reply scripts/worktree_setup.py]: Fixed in commit 265b07ef — branch name is now wrapped with shlex.quote() before being interpolated into the push-reminder message. —Claude

File: scripts/tests/test_worktree_setup.py [hunk 6]
```diff
@@ -528,5 +531,186 @@ def test_trailing_special_chars_stripped(self) -> None:
...
+            mock_run.return_value = self._make_completed(
+                returncode=0, stdout="origin/feat/my-feature\n"
+            )
+            configure_branch_tracking("feat/my-feature", tmp_path)
+        # Only one git call: the @{u} lookup, then remote_check, then early return
+        # (remote exists but upstream is already correct → no set-upstream call)
+        set_upstream_calls = [
+            c for c in mock_run.call_args_list if "--set-upstream-to" in str(c)
+        ]
+        assert len(set_upstream_calls) == 0
```
[the remark this answers is not in the document]
Comment (timothyfroehlich) [reply scripts/tests/test_worktree_setup.py]: Declined per the testing philosophy now documented at the top of this test file: worktree setup is infrastructure code we test by running it. The pruned-ref edge case is rare and observable on first push (the user sees the reminder, runs `git fetch && git push -u`, problem fixed). Adding mock-heavy unit tests for git interactions trades real-world coverage for testing the mock harness. —Claude

File: scripts/worktree_setup.py [hunk 7]
```diff
@@ -511,6 +511,89 @@ def get_branch() -> str:
...
+        == 0
+    )
+
+    if not has_remote:
+        print(
+            f"worktree_setup: '{branch}' has no remote yet — "
+            f"run `git push -u origin {branch}` on first push",
+            file=sys.stderr,
+        )
+        return
```
[the remark this answers is not in the document]
Comment (timothyfroehlich) [reply scripts/worktree_setup.py:576]: Fixed in commit 265b07ef — when origin/<branch> doesn't exist yet and the current upstream is origin/main or origin/master, we now run `git branch --unset-upstream` before printing the reminder, so `git pull` no longer pulls from main in the new worktree. —Claude
